File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String smStr = "";
        int min = Integer.MAX_VALUE;
        for(int i=0;i<strs.length;i++){
            if(strs[i].length() < min){
                min = strs[i].length();
                smStr = strs[i];
            }
        }
        String ans = "";
        for(int i=0; i<smStr.length();i++){
            boolean pre = true;
            for(int j=0;j<strs.length;j++){
                if(smStr.charAt(i) != (strs[j].charAt(i))){
                    pre = false;
                    break;
                }
            }
            if(pre){
                ans+=""+smStr.charAt(i);
            }else{
                break;
            }
        }
        return ans;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if len(strs) == 1:
            return strs[0]
        ans = strs[0]
        for s in strs[1:]:
            check, s = ans if len(ans) <= len(s) else s, s if len(ans) <= len(s) else ans
            i=0
            while i < len(check) and check[i] == s[i]:
                i+=1
            ans = check[:i]
        return ans